src/methods/tx/_events.ts

```diff
@@ -0,0 +1,345 @@
+         * and the second amount is asset B going from second account to first account.
+         *
+         * Which assets have been used - check in event "IntentionRegistered" by
+         * appropriate IntentionID.
+         *
+         * One exchange action (sell/buy) can includes multiple direct trade transactions,
+         * that's why we need track all direct trade transactions for one exchange action.
+         */
+        if (Array.isArray(parsedData)) {
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            intentions: [parsedData[2]?.toString(), parsedData[3]?.toString()],
+            data: {
+              id: parsedData[2]?.toString(),
+              directTrades: [
+                {
+                  amountSent: new BigNumber(parsedData[4]?.toString() || 0),
+                  amountReceived: new BigNumber(parsedData[5]?.toString() || 0),
+                  account1: parsedData[0]?.toString(),
+                  account2: parsedData[1]?.toString(),
+                  pairedIntention: parsedData[3]?.toString(),
+                },
+              ],
+            },
+          });
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            intentions: [parsedData[2]?.toString(), parsedData[3]?.toString()],
+            data: {
+              id: parsedData[3]?.toString(),
+              directTrades: [
+                {
+                  amountSent: new BigNumber(parsedData[5]?.toString() || 0),
+                  amountReceived: new BigNumber(parsedData[4]?.toString() || 0),
+                  account1: parsedData[1]?.toString(),
+                  account2: parsedData[0]?.toString(),
+                  pairedIntention: parsedData[2]?.toString(),
+                },
+              ],
+            },
+          });
+        }
+        break;
+      case 'IntentionResolvedDirectTradeFees':
+        /**
+         * parsedData: <Array> [AccountId, AccountId, AssetId, Balance]
+         *                     [who, account paid to, asset, fee amount]
+         */
+        break;
+      case 'IntentionResolveErrorEvent':
+        /**
+         * parsedData: <Array> [AccountId, AssetPair, IntentionType, IntentionId, dispatch]
+         *                     [who, assets, sell or buy, intention id, error detail]
+         */
+        if (Array.isArray(parsedData)) {
+          mergeEventToScope({
+            ...exchangeTxEventData,
+            status: {
+              ...exchangeTxEventData.status,
+              error: [
+                {
+                  method,
+                  data,
+                },
+              ],
+            },
+            data: {
+              id: parsedData[3]?.toString(),
+              account: parsedData[0]?.toString(),
+              intentionType: parsedData[2]?.toString(),
+              assetsPair: parsedData[1]?.toString(),
+              errorDetails: parsedData[1]?.toString(),
```

Comment:
assetsPair and errorDetails are from same source `parsedData[1]`.
Is it correct behavior? And if it's correct, would it better to conditionally assign values? I guess there is no case that assetsPair and errorDetails will be set simultaneously.

Reply:
it's my typo. I'll fix it.